src/harness/BenchmarkDotNet.Extensions/PartitionFilter.cs

```diff
@@ -0,0 +1,32 @@
+using BenchmarkDotNet.Filters;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Running;
+
+
+public class PartitionFilter : IFilter
+{
+    private readonly int? _partitionsCount;
+    private readonly int? _partitionIndex; // indexed from 0
+    private int _counter = 0;
+ 
+    public PartitionFilter()
+    {
+        _partitionsCount = ReadFromEnvironmentVariable("PARTITIONS_COUNT");
```

Comment:
I'd like it to take command line parameters instead of relying on the environment. This means a more complex change but I think that's goodness.

Reply:
Done.